cmd/warpforge/main_test.go

```diff
@@ -82,12 +94,29 @@ func cleanRunRecord(str string) string {
 	return strings.TrimSpace(str)
 }
 
-func execFn(args []string, stdin io.Reader, stdout, stderr io.Writer) (int, error) {
-	err := makeApp(stdin, stdout, stderr).Run(args)
-	if err != nil {
-		return 1, err
+func buildExecFn(projPath string) func(args []string, stdin io.Reader, stdout io.Writer, stderr io.Writer) (int, error) {
+	return func(args []string, stdin io.Reader, stdout, stderr io.Writer) (int, error) {
+		// override the path to required binaries
+		err := os.Setenv("WARPFORGE_PATH", filepath.Join(projPath, "plugins"))
+
+		// set up a root workspace in the testmark run directory
+		testmarkWd, err := os.Getwd()
+		if err != nil {
+			return 1, fmt.Errorf("failed to get testmark pwd: %s", err)
+		}
+		os.MkdirAll(filepath.Join(testmarkWd, ".warpforge/catalogs/default"), 0755)
+		copyCmd := exec.Command("cp", "--recursive", filepath.Join(projPath, ".warpforge", "catalog"), filepath.Join(testmarkWd, ".warpforge", "catalogs"))
```

Comment:
How many pieces of data are we moving around with this?  Are they textual, such that we could stick them in a data hunk in testmark, and thus not need this operation?

Still thinking about exactly how that would work, but... anything we can do to make this piece of setup less magical, and more legible in docs and such, would probably be a win.